--- ../index.php        2017-10-01 21:04:22.894526997 +0000
+++ index.php   2017-10-01 21:01:36.438624343 +0000
@@ -131,6 +131,37 @@

         document.location = new_location;
       }
+
+       function loadFrame(frame) {
+               var xhr = new XMLHttpRequest();
+               xhr.onreadystatechange = function() {
+                   if (xhr.readyState == XMLHttpRequest.DONE) {
+                       if (xhr.status == 200) {
+                               frame.style.backgroundColor = 'green';
+                               frame.contentWindow.document.open();
+                               frame.contentWindow.document.write(xhr.responseText);
+                               frame.contentWindow.document.close();
+                       } else {
+                               frame.style.backgroundColor = 'red';
+                               setTimeout(function() { loadFrame(frame); }, 500);
+                       }
+                   }
+               }
+               if (Math.random() > 0.7) {
+                       xhr.open('GET', 'badwolf', true);
+               } else {
+                       xhr.open('GET', 'frame.php?refresh=<?= $_GET['refresh'] ?>', true);
+               }
+               xhr.send(null);
+
+       }
+
+       function pageLoaded() {
+               var frames = document.getElementsByTagName('IFRAME');
+               for (i = 0; i < frames.length; i++) {
+                       loadFrame(frames[i]);
+               }
+       }
     </script>
   </head>
 <?php
@@ -143,7 +174,7 @@
     $rows = 1;
   }
 ?>
-  <body>
+  <body onload='pageLoaded()'>
     <div id='banner'>
       <img id='gcp' src='google-cloud-platform-logo.png'>
       <span id='amp'>+</span>
@@ -182,7 +213,7 @@
         <tr>
 <?php   for ($i = 0; $i < $cols; $i++) { ?>
           <td>
-            <iframe src='frame.php?refresh=<?= $_GET['refresh'] ?>'></iframe>
+            <iframe></iframe>
           </td>
 <?php   } ?>
         </tr>
