diff -u ../frame.php ./frame.php
--- ../frame.php        2017-10-01 20:18:34.776241805 +0000
+++ ./frame.php 2017-10-01 21:24:16.601805371 +0000
@@ -17,9 +17,6 @@
   <meta http-equiv="Cache-Control" content="no-cache, no-store" />
   <meta http-equiv="Pragma" content="no-cache" />
   <meta http-equiv="Expires" content="0" />
-<?php if ($_GET['refresh']) { ?>
-  <meta http-equiv="refresh" content="5">
-<?php } ?>
   <style>
     body {
       margin: 0;
@@ -58,25 +55,32 @@
       text-align: center;
       padding: 10px;
     }
-    #refresh {
-      font-family: Roboto;
-      font-size: calc(14px + 1vw);
-    }
-    #refresh A {
-      text-decoration: none;
+    #loaded {
+      position: absolute;
+      font-size: 24pt;
+      font-weight: bold;
+      bottom: 4pt;
+      right: 4pt;
+      color: white;
+      opacity: 0.5;
     }
   </style>
   <title><?= $host ?></title>
+  <script>
+    function flashReload() {
+      setTimeout(function() {
+        document.all.loaded.style.display = 'none';
+      }, 1000);
+    }
+  </script>
 </head>
-<body>
+<body onload='flashReload()'>
   <div class="frameT">
     <div class="frameTC">
       <div class="content">
         <h1><script>document.write(document.title);</script></h1>
-        <?= date('r') ?>
-        <span id='refresh'>
-          <a href='javascript:location.reload(true)'><?= $_GET['refresh'] ? '&#x21bb;' : '' ?></a>
-        </span>
+        <span id='generated'><?= date('r') ?></span>
+       <div id='loaded'>&#x21bb;</div>
       </div>
     </div>
   </div>
diff -u ../index.php ./index.php
--- ../index.php        2017-10-01 21:08:17.334387652 +0000
+++ ./index.php 2017-10-01 21:27:29.393685447 +0000
@@ -132,6 +132,21 @@
         document.location = new_location;
       }

+      function scheduleFrameReload() {
+        setTimeout('reloadFrames()', 5000);
+      }
+
+      function reloadFrames() {
+        loadFrames();
+        scheduleFrameReload(); // again
+      }
+
+      function loadFrames() {
+        var frames = document.getElementsByTagName('IFRAME');
+        for (i = 0; i < frames.length; i++) {
+          loadFrame(frames[i]);
+        }
+      }
       function loadFrame(frame) {
         var xhr = new XMLHttpRequest();
         xhr.onreadystatechange = function() {
@@ -141,21 +156,23 @@
               frame.contentWindow.document.write(xhr.responseText);
               frame.contentWindow.document.close();
             } else {
-              frame.style.backgroundColor = 'red';
+              frame.contentWindow.document.open();
+              frame.contentWindow.document.write("<font size='18pt'>&#x2620;</font>");
+              frame.contentWindow.document.close();
               setTimeout(function() { loadFrame(frame); }, 500);
             }
           }
         }
-        xhr.open('GET', 'frame.php?refresh=<?= $_GET['refresh'] ?>', true);
+        xhr.open('GET', 'frame.php', true);
         xhr.send(null);
-       }
+      }

-       function pageLoaded() {
-         var frames = document.getElementsByTagName('IFRAME');
-         for (i = 0; i < frames.length; i++) {
-           loadFrame(frames[i]);
-         }
-       }
+      function pageLoaded() {
+        loadFrames();
+<?php if ($_GET['refresh']) { ?>
+        scheduleFrameReload();
+<?php } ?>
+      }
     </script>
   </head>
 <?php
