--- ../index.php        2017-10-01 22:04:33.992322642 +0000
+++ ./index.php 2017-10-01 22:09:40.944137452 +0000
@@ -16,6 +16,9 @@
   }
   $failures = $_GET['failures'];
   $failure_sec = $_GET['failure_sec'];
+  if (empty($failure_sec)) {
+    $failure_sec = 2;
+  }
 ?>
 <html>
   <head>
@@ -235,7 +238,7 @@
       <select id='failure_sec' name='failure_sec' onchange='updateLoad()'
           <?= ! $failures ? 'disabled' : '' ?>>
 <?php foreach ([0.1, 0.5, 1, 2] as $v) { ?>
-        <option <?= $failure_sec == $v ? 'selected' : ''?> value='<?= $v ?>'><?= $v ?> sec</option>
+        <option <?= $failure_sec == $v ? 'selected' : ''?> value='<?= $v ?>'><?= $v ?> s</option>
 <?php } ?>
       </select>
     </div>
